zio-http/src/main/scala/zio/http/api/HttpCodec.scala

```diff
@@ -210,8 +208,11 @@ object HttpCodec extends HeaderCodecs with QueryCodecs with RouteCodecs {
     def erase: Method[Any] = self.asInstanceOf[Method[Any]]
   }
 
-  private[api] final case class Header[A](name: String, textCodec: TextCodec[A], optional: Boolean)
-      extends Atom[CodecType.Header, A] {
+  private[api] final case class Header[A](
+    name: String,
+    textCodec: Either[TextCodec[A], RichTextCodec[A]],
```

Comment:
This is good for now, eventually we want to delete the `Either` and just use `RichTextCodec` because the other one is too weak for documentation and client purposes.